pages/api/openai.js

```diff
@@ -0,0 +1,56 @@
+import { OpenAIApi, Configuration } from "openai";
+import { ExerciseFormat } from "../../lib/data-format";
+import ExerciseFormResult from "../../components/ExerciseFormResult";
+
+export default async function handler(req, res) {
+  const configuration = new Configuration({
+    apiKey: process.env.OPENAI_API_KEY,
+  });
+  const openai = new OpenAIApi(configuration);
+
+  try {
+    const response = await openai.createChatCompletion({
+      model: "gpt-3.5-turbo",
+      messages: [
+        {
+          role: "system",
+          content:
+            "Du übernimmst die Rolle eines kompetenten Assistenten, der auf personalisierte Anfragen Trainingsempfehlungen gibt. Antworte bitte im angegebenen JSON-Format und beachte die Anweisungen des Benutzers sorgfältig. Deine Antworten sollten präzise, hilfreich und auf die Erstellung eines Trainingsplans mit Eigengewichtsübungen für Anfänger in deutscher Sprache ausgerichtet sein.",
+        },
+        {
+          role: "user",
+          content: `Bitte erstelle einen maßgeschneiderten Trainingsplan für Anfänger in deutscher Sprache.
+          Anhand der vorgebenen Fragen und Antworten:
+          ${ExerciseFormResult}
+          Wähle 15 Eigengewichtsübungen aus, die du empfiehlst.
+          Für jede dieser Übungen benötige ich folgende Angaben:
+          - Übungsname
+          - Kurze Übersicht der Übung
+          - Anzahl der Wiederholungen pro Satz
+          - Anzahl der Sätze
+          - YouTube URl zu einem Anleitungs oder Tutorial Video in Deutsch zur Übung (Bitte achte darauf das das video verfügbar ist)
+          Deine Antwort sollte genau dem folgenden JSON-Format entsprechen:
+          ${JSON.stringify(ExerciseFormat)}
+          Bitte beschränke dich auf das angegebene Format und verzichte auf zusätzlichen Text in der Antwort und achte auf eine korrekte Json Syntax! Bitte bennene das Array "exercises"`,
+        },
+      ],
+      temperature: 0,
+      max_tokens: 3000,
+      top_p: 0,
+      frequency_penalty: 0,
+      presence_penalty: 0,
+    });
+    const progress = Math.min(
+      Math.round((response.data.usage.total_tokens / 3000) * 100)
+    );
```

Comment:
Don't understand the usage of Math.min here since there is only one Number inside. (???)